### PR TITLE
Fix stream playlists as directory

### PR DESCRIPTION
**Pull request by vrichter**
## Description
Workaround for broken STRM playback from favorites after #14183

## Motivation and Context
Patch #14183 introduced a regression, preventing playlists with streams from being played correctly.
This fixes the problem.,

## How Has This Been Tested?
Ubuntu 18.04, self built master. Tested playback of local playlist and playlist with stream. 

## Types of change
- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by vrichter** [comment]
This patch does not change much but requires the indentation of a block of code. That is why it looks as if much was changed. The first commit holds the changes in code. The second is only cosmetic.

As a drawback: This solution still tries to expand the playlists as a directory.
In the case of stream files this results in a ``not a directory`` error being logged.

File: xbmc/interfaces/builtins/PlayerBuiltins.cpp [hunk 1]
```diff
@@ -435,39 +435,43 @@ static int PlayMedia(const std::vector<std::string>& params)
     std::string extensions = CServiceBroker::GetFileExtensionProvider().GetVideoExtensions() + "|" + CServiceBroker::GetFileExtensionProvider().GetMusicExtensions();
     XFILE::CDirectory::GetDirectory(item.GetPath(), items, extensions, XFILE::DIR_FLAG_DEFAULTS);
 
-    bool containsMusic = false, containsVideo = false;
-    for (int i = 0; i < items.Size(); i++)
+    if (items.Size() != 0) // fall through on non expandable playlist
```
Comment (Rechi) [inline xbmc/interfaces/builtins/PlayerBuiltins.cpp]: `!items.IsEmpty()`

**Comment by notspiff** [comment]
wanna take a stab at removing the historic error that strm files are playlists?

**Review by notspiff — APPROVED** [review]

**Comment by vrichter** [comment]
squashed and rebased.

**Comment by vrichter** [comment]
> wanna take a stab at removing the historic error that strm files are playlists?
I can look into it if i find the time but I cannot estimate the side effects of such a change.

Nonetheless, this fix is still needed because m3u playlists can be lists of streams too.

The strm problem should be a different pull-request.

**Comment by notspiff** [comment]
Absolutely, i have already approved this..

**Comment by FernetMenta** [comment]
> wanna take a stab at removing the historic error that strm files are playlists?

strm files can be anything, including playlists, i.e. a strm file can contain a m3u URL

**Comment by FernetMenta** [comment]
you are right, the strm by its own can never be a playlist. sorry for the noise

**Pull request merged**